### ISSUE TITLE
Invalid region provided: ap-south-1

### ISSUE THREAD

**Issue by sbobylev** [docker/distribution#2498]
The registry fails to start when I attempt run it using the S3 storage backend and a bucket in the ap-south-1 region.

Very similar issue was reported in   #2058

My registry version: **2.6**

Command:

```
docker run --rm -p 5000:5000 --name registry \
                        -e REGISTRY_STORAGE=s3 \
                        -e REGISTRY_STORAGE_S3_REGION=ap-south-1 \
                        -e REGISTRY_STORAGE_S3_BUCKET=my-bucket-name \
                        registry:2
```

Error:

```
panic: Invalid region provided: ap-south-1

goroutine 1 [running]:
panic(0xb7dbc0, 0xc4203fb240)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/docker/distribution/registry/handlers.NewApp(0x1066800, 0xc420398060, 0xc420133180, 0x1066800)
	/go/src/github.com/docker/distribution/registry/handlers/app.go:123 +0x3908
github.com/docker/distribution/registry.NewRegistry(0x7f8dd3b853d8, 0xc420398060, 0xc420133180, 0xc420133180, 0x0, 0x0)
	/go/src/github.com/docker/distribution/registry/registry.go:86 +0x213
github.com/docker/distribution/registry.glob..func1(0x108e180, 0xc420370c20, 0x1, 0x1)
	/go/src/github.com/docker/distribution/registry/registry.go:55 +0x106
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).execute(0x108e180, 0xc420370bd0, 0x1, 0x1, 0x108e180, 0xc420370bd0)
	/go/src/github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:495 +0x190
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).Execute(0x108e320, 0xc4201e7f40, 0xc4200001a0)
	/go/src/github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:560 +0x3c3
main.main()
	/go/src/github.com/docker/distribution/cmd/registry/main.go:24 +0x2d

```

**Comment by sbobylev** [comment]
Since the region is not checked when an endpoint is provided as per this code block 

```
	if params.RegionEndpoint != "" {
		awsConfig.WithS3ForcePathStyle(true)
		awsConfig.WithEndpoint(params.RegionEndpoint)
        }
```

I can pass the following variable as a workaround ```REGISTRY_STORAGE_S3_REGIONENDPOINT=s3.ap-south-1.amazonaws.com``` for now.

**Comment by dmcgowan** [comment]
This is closed by #2140, we currently have no release time frame for 2.7 but you can apply this change to the 2.6 branch or use master
